module/documents/actor/actor.mjs

```diff
@@ -2870,7 +2870,16 @@ export default class Actor5e extends SystemDocumentMixin(Actor) {
       if ( settings.keep.has("languages") ) d.system.traits.languages = o.system.traits.languages;
 
       // Keep specific items from the original data
-      d.items = d.items.concat(o.items.filter(i => {
+      const profDiff = source.system.attributes.prof - this.system.attributes.prof;
+      d.items = d.items.map(i => {
+        if ( settings.keep.has("class") && ((i.type === "feat") || (i.type === "weapon")) ) {
+          // Items gained from the source should use the source's proficiency bonus.
+          Object.values(i.system.activities).forEach(activity => {
+            if ( activity.type === "attack" ) activity.attack.bonus += `${profDiff < 0 ? "" : "+"}${profDiff}`;
```

Comment:
Space needed in case the existing bonus ends in a `@` value (`@prof-13` won't evaluate properly):

```suggestion
            if ( activity.type === "attack" ) activity.attack.bonus += ` ${profDiff < 0 ? "" : "+"}${profDiff}`;
```